### ISSUE TITLE
Crash when getter for error.name throws an error

### ISSUE THREAD

**Issue by Yogu** [arangodb/arangodb#4308]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.2
- [x] 3.3

Mode:
- [x] Single-Server

Storage-Engine:
- [x] mmfiles

On this operating system:
- [x] Windows, version: 10

I'm issuing AQL via:
- [x] arangojs

When you run a JavaScript snippet like the following:

```javascript
var err = new Error('test');
Object.defineProperty(err, 'name', {
  get: function() { throw new Error('Error in getter'); }
});
throw err;
```

arangod crashes without notice (debug log levle enabled) and needs to recover when started the next time.

Well-behaved code obviously should not throw there, but arangodb also should not crash, I think.

**Comment by jsteemann** [comment]
How do you execute the above code exactly in ArangoDB?

**Comment by Yogu** [comment]
I pass the JS code to `db.transaction()`.

**Comment by jsteemann** [comment]
I am trying to reproduce this using the following code from the ArangoShell, but with no luck yet:
```
db._executeTransaction({ collections: { read: "a" }, action: function() {   var err = new Error('test'); Object.defineProperty(err, 'name', {   get: function() { throw new Error('Error in getter'); } }); throw err; } });
```
Can you share the exact command you are using so we can reproduce it?

**Comment by Yogu** [comment]
I, I just executed exactly your snippet (with collection "a" replaced by "_modules") and it crashed the ArangoDB service. Did you test it on Windows 10? Also, what wa your result? How was the error reported?

**Comment by jsteemann** [comment]
Tried this with both 3.2 and 3.3 (latest versions of the respective branches) with a single server instance on Linux using the MMFiles engine.
The error message in both cases was:
```
JavaScript exception in file 'js/client/modules/@arangodb/arango-database.js' at 1129,5: ArangoError 1650: transaction not running on commit
!    throw new ArangoError(requestResult);
!    ^
stacktrace: ArangoError: transaction not running on commit
    at ArangoDatabase._executeTransaction (js/client/modules/@arangodb/arango-database.js:1129:11)
    at <shell command>:1:4
```
Trying the same on Windows produced the crash.
Looking into it now.

**Comment by jsteemann** [comment]
Fixed in 3.2.11 and 3.3.3

**Comment by maxkernbach** [comment]
Version 3.2.11 has been released and includes the fix for the problem you reported.
Please reopen this issue if the problem still occurs.

Thanks
